cloud/user_data.go

```diff
@@ -130,33 +135,59 @@ func parseUserDataContentType(userData string) (string, error) {
 	return "", errors.Errorf("user data format is not recognized from first line: '%s'", firstLine)
 }
 
-// bootstrapUserData returns the multipart user data with logic to bootstrap and
-// set up the host and the custom user data. Care should be taken when adding
-// more to this script, since the script length is subject to a 16 kB hard limit
+// bootstrapUserData returns the user data with logic to bootstrap and set up
+// the host and the custom user data.
+// If mergeParts is true, the bootstrap part and the custom part will
+// be combined into a single user datapart, so they must be the same type (e.g.
+// if shell scripts, both must be the same shell scripting language).
+// Care should be taken when adding more content to user data, since the user
+// data length is subject to a 16 kB hard limit
 // (https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#RunInstancesInput).
-func bootstrapUserData(ctx context.Context, env evergreen.Environment, h *host.Host, customScript string) (string, error) {
+// We could possibly increase the length by gzip compressing it.
+func bootstrapUserData(ctx context.Context, env evergreen.Environment, h *host.Host, custom string, mergeParts bool) (string, error) {
 	if h.Distro.BootstrapSettings.Method != distro.BootstrapMethodUserData {
-		return customScript, nil
+		return custom, nil
 	}
 	settings := env.Settings()
 
 	creds, err := h.GenerateJasperCredentials(ctx, env)
 	if err != nil {
-		return customScript, errors.Wrapf(err, "problem generating Jasper credentials for host '%s'", h.Id)
+		return custom, errors.Wrapf(err, "problem generating Jasper credentials for host '%s'", h.Id)
```

Comment:
why return the script and an error?

Reply:
It doesn't really matter what you return here, but I changed it to return "" anyways.